### PR TITLE
[Backport] 8266317: Vector API enhancements

### PR DESCRIPTION
**Pull request by JinZhonghui**
Summary: [Backport] 8266317: Vector API enhancements

Test Plan: ci jtreg

Reviewed-by: JoshuaZhuwj

Issue: https://github.com/dragonwell-project/dragonwell11/issues/617

### CODE REVIEW

**Comment by JinZhonghui** [comment]
@JoshuaZhuwj Please help review it.

**Review by JoshuaZhuwj — COMMENTED** [review]
Thanks a lot for your change.
Overall it looks good to me.
Just a minor comment.

File: src/hotspot/share/opto/subnode.hpp [hunk 1]
```diff
@@ -280,10 +280,15 @@ class CmpD3Node : public CmpDNode {
 // Convert condition codes to a boolean test value (0 or -1).
 // We pick the values as 3 bits; the low order 2 bits we compare against the
 // condition codes, the high bit flips the sense of the result.
+// For vector compares, additionally, the 4th bit indicates if the compare is unsigned
 struct BoolTest {
-  enum mask { eq = 0, ne = 4, le = 5, ge = 7, lt = 3, gt = 1, overflow = 2, no_overflow = 6, illegal = 8 };
+  enum mask { eq = 0, ne = 4, le = 5, ge = 7, lt = 3, gt = 1, overflow = 2, no_overflow = 6, illegal = 8,
```
[the remark this answers is not in the document]
Comment (JinZhonghui) [reply src/hotspot/share/opto/subnode.hpp:285]: The 'nerver' was introduced from JDK-8210215, but we didn't have it.

[the remark this answers is not in the document]
Comment (JoshuaZhuwj) [reply src/hotspot/share/opto/subnode.hpp:285]: Understood. Thanks!

**Review by JoshuaZhuwj — APPROVED** [review]

**Pull request merged**